src/espic/solve_maxwell.py

```diff
@@ -34,3 +35,103 @@ def solve(self, rho):
         phi[1:-1] = solve_banded((1, 1), bands, rhs)
 
         return phi
+
+
+# Code taken from https://john-s-butler-dit.github.io/NumericalAnalysisBook/Chapter%2009%20-%20Elliptic%20Equations/903_Poisson%20Equation-Boundary.html
+# FIXME: for now, this assumes equal spacing in x and y.
+class MaxwellSolver2D:
+    def __init__(self, grid=Uniform2DGrid(), boundary_conditions=None):
+        self.grid = grid.grid
+        self.xgrid = grid.xgrid
+        self.ygrid = grid.ygrid
```

Comment:
You might consider slightly changing the names of some variables to be consistent. For instance, `xgrid` -> `x_grid`, `ygrid` -> `y_grid`, `gridsize` -> `grid_size`, etc.

Reply:
yeah I'll chase these down, thanks for the reminder.